src/main/java/me/decryptedx/dcxshulkerlimiter/DCXPermissionLimitExtractor.java

```diff
@@ -0,0 +1,43 @@
+package me.decryptedx.dcxshulkerlimiter;
+
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.permissions.PermissionAttachmentInfo;
+
+import java.util.logging.Level;
+
+/**
+ * Helper class that encapsulates getting a player's highest effective permission for a permission node
+ * that starts with a prefix
+ */
+public class DCXPermissionLimitExtractor {
+    private final Player player;
+
+    public DCXPermissionLimitExtractor(final Player player) {
+        this.player = player;
+    }
+
+    /**
+     *
+     * @param prefix Permission node prefix
+     * @param defaultValue Default value of permission
+     * @return Highest value of permission nodes in the player's permission with a given prefix
+     */
+    public int getLimit(String prefix, int defaultValue) {
```

Comment:
I like the addition of this method, but we can avoid needing to initialize an object by making the method static and passing player as a parameter.

Reply:
Will convert into method after merging.